lib/phase/flatten.js

```diff
@@ -30,6 +30,42 @@ function merge(newJob, oldJob) {
     const oldSecrets = oldJob.secrets || [];
 
     newJob.secrets = [...new Set([...newSecrets, ...oldSecrets])];  // remove duplicate
+
+    if (fromTemplate && oldJob.steps) {
+        let stepName;
+        let preStepName;
+        let postStepName;
+        const mergedSteps = [];
+
+        // convert steps from oldJob from array to object for faster lookup
+        const oldSteps = oldJob.steps.reduce((obj, item) => {
+            const key = Object.keys(item)[0];
```

Comment:
This is the only bit I don't understand. What do the elements of `oldJob.steps` look like?